opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -87,264 +88,282 @@ message InstrumentationLibraryMetrics {
-
-    // Summary value. Some frameworks implemented Histograms as a summary of observations
-    // (usually things like request durations and response sizes). While it
-    // also provides a total count of observations and a sum of all observed
-    // values, it calculates configurable percentiles over a sliding time
-    // window.
-    // Corresponding values are stored in SummaryDataPoint.
-    SUMMARY = 6;
+  // ValueType is the type of values a metric has.  ValueType determines
+  // which field of the DataPoint will be used for Metrics with this
+  // descriptor.
+  enum ValueType {
+    // INVALID_TYPE is the default ValueType, it MUST not be used.
+    INVALID_VALUE_TYPE = 0;
+
+    // Note that int64 ValueTypes use the even numbers and double
+    // ValueTypes use the odd numbers.
+
+    // Metric.Points MUST use DataPoint.value_int64.
+    SCALAR_INT64     = 2;
+    // Metric.Points MUST use DataPoint.value_double.
+    SCALAR_DOUBLE    = 3;
+
+    // Metric.Points MUST use DataPoint.histogram with int64 boundaries.
+    HISTOGRAM_INT64  = 4;
+    // Metric.Points MUST use DataPoint.histogram with floating point boundaries.
+    HISTOGRAM_DOUBLE = 5;
+
+    // Metric.Points MUST use DataPoint.summary with int64 min/max/sum/last values.
+    SUMMARY_INT64    = 6;
+    // Metric.Points MUST use DataPoint.summary with double min/max/sum/last values.
+    SUMMARY_DOUBLE   = 7;
+
+    // Metric.Points MUST use DataPoint.raw with int64 values.
+    RAW_INT64        = 8;
+    // Metric.Points MUST use DataPoint.raw with double value.
+    RAW_DOUBLE       = 9;
   }
 
-  // type is the type of values this metric has.
-  Type type = 4;
+  // value_type is the type of values this metric has.
+  ValueType value_type = 4;
 
-  // Temporality is the temporal quality values of a metric have. It
-  // describes how those values relate to the time interval over which they
-  // are reported.
-  enum Temporality {
-    // INVALID_TEMPORALITY is the default Temporality, it MUST not be
-    // used.
-    INVALID_TEMPORALITY = 0;
+  // KindMask contains a set of bit masks used to construct Kind enum values.
+  // There are 7 bits used presently.
+  enum KindMask {
+    // INVALID_KIND_MASK is not used.
+    INVALID_KIND_MASK = 0;
+
+    // One of the following three MUST be set. There are 3 exclusive Temporality kinds.
 
     // INSTANTANEOUS is a metric whose values are measured at a particular
     // instant. The values are not aggregated over any time interval and are
-    // unique per timestamp. As such, these metrics are not expected to have
+    // unique per timestamp. As such, these metrics are not expected to have	
     // an associated start time.
-    INSTANTANEOUS = 1;
+    INSTANTANEOUS = 0x1;
 
     // DELTA is a metric whose values are the aggregation of measurements
     // made over a time interval. Successive metrics contain aggregation of
     // values from continuous and non-overlapping intervals.
-    //
+    //	
```

Comment:
these odd whitespace changes make for a slightly messy diff. any reason why these showed up?

Reply:
I was working to restore this. I had removed all the comments for some clarity, and now am deciding on a case-by-case basis whether they should be updated (making a larger diff) or left alone. Many comments are slightly stale. I'll keep working on this.